File: ContainerWithMostWater.cs
using System;
namespace Algorithms
{
    public class ContainerWithMostWater
    {
        /// Time Complexity : O(N)
        // Space Complexity :O(1)
        // Did this code successfully run on Leetcode :Yes
        // Any problem you faced while coding this :  Based on the discussion in class
        public int MaxArea(int[] height)
        {

            if (height == null || height.Length == 0) return 0;
            int n = height.Length;
            int left = 0;
            int right = n - 1;
            int max = 0;
            while (left < right)
            {
                max = Math.Max(max, Math.Min(height[left], height[right]) * (right - left));
                if (height[left] <= height[right])
                {
                    left++;
                }
                else
                {
                    right--;
                }
            }

            return max;
        }
    }
}

File: ContainerWithMostWater.js
// Time Complexity : O(n)
// Space Complexity : O(1)
// Did this code successfully run on Leetcode : Yes
// Any problem you faced while coding this : No

/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
  if (height.length < 2 || height === null) return 0;

  var n = height.length;
  var low = 0;
  var high = n - 1;
  var max = 0;

  while (low < high) {
    var width = high - low;
    var minHeight = Math.min(height[low], height[high]);
    max = Math.max(max, minHeight * width);
    if (height[low] < height[high]) {
      low++;
    } else {
      high--;
    }
  }
  return max;
};
